### PR TITLE
Added deprecated.jl

### PR DESCRIPTION
**Pull request by rjplevin**
Added some deprecations to assist with converting models.

### CODE REVIEW

**Comment by lrennels** [comment]
Good idea!  From the integration doc I think the list of most commonly used deprecated functions is probably the table below, although they aren't the only ones that changed.  These could be added to the deprecated file if that seems wise to you. 

| Old Syntax                | New Syntax                |
| ------------------------  |:-------------------------:|
|`addcomponent!`            |`add_comp!`                |
|`connectparameter`         |`connect_param!`           |
|`setleftoverparameters`    |`set_leftover_params!`     |
|`setparameter`             |`set_param!`           |
|`adddimension`             |`add_dimension!`           |
|`setindex`                 |`set_dimension!`           |  
|`isstart`                  |`is_first`                 |
|`isstop`                   |`is_last`                  |

File: src/core/deprecated.jl [hunk 1]
```diff
@@ -0,0 +1,17 @@
+using Base: @deprecate
+
+@deprecate addcomponent add_comp!
+@deprecate set_parameter! set_param!
+
+@deprecate connect_parameter(md::ModelDef, comp_name::Symbol, param_name::Symbol, ext_param_name::Symbol) connect_param!(md, comp_name, param_name, ext_param_name)
+
+@deprecate(
+    connect_parameter(md::ModelDef, dst_comp_name::Symbol, dst_par_name::Symbol, src_comp_name::Symbol, src_var_name::Symbol, backup::Union{Void, Array}=nothing; ignoreunits::Bool=false, offset::Int=0),
+    connect_parameter(md, dst_comp_name, dst_par_name, src_comp_name, src_var_name, backup; ignoreunits=ignoreunits, offset=offset)
```
Comment (lrennels) [inline src/core/deprecated.jl]: should the new function be `connect_param!`?

Comment (rjplevin) [reply src/core/deprecated.jl]: Yes!

Comment (lrennels) [reply src/core/deprecated.jl]: On it!

**Pull request closed without merging**